micronaut-build-plugin-sourcegen-generator/src/test/groovy/io/micronaut/sourcegen/generator/visitors/GradlePluginGenerationSpec.groovy

```diff
@@ -0,0 +1,209 @@
+  }
+
+  public static class WolfClasspathConfigurator implements Action<ClassLoaderWorkerSpec> {
+    WolfTask task;
+
+    public WolfClasspathConfigurator(WolfTask task) {
+      this.task = task;
+    }
+
+    public void execute(ClassLoaderWorkerSpec spec) {
+      spec.getClasspath().from(this.task.getClasspath());
+    }
+  }
+}"""
+
+        var specContent = stripImports(files.get("test.WolfSpec").getCharContent(false))
+        specContent == """/**
+ * Specification that is used for configuring Wolf task.
+ * Wolf Gradle task.
+ */
+public interface WolfSpec {
+  /**
+   * Configurable slogan parameter.
+   */
+  Property<String> getSlogan();
+
+  /**
+   * Configurable age parameter.
+   */
+  Property<Integer> getAge();
+}"""
+
+        var extensionContent = stripImports(files.get("test.WolfExtension").getCharContent(false))
+        extensionContent == """/**
+ * Configures the Wolf execution.
+ */
+public interface WolfExtension {
+  /**
+   * Create a task for awooo.
+   * Main execution of Wolf task.
+   * @param name The unique identifier used to derive task names
+   * @param spec The configurable specification
+   */
+  void awooo(String name, Action<? super WolfSpec> action);
+}"""
+
+        var defaultExtensionContent = stripImports(files.get("test.DefaultWolfExtension").getCharContent(false))
+        defaultExtensionContent == """public abstract class DefaultWolfExtension implements WolfExtension {
+  protected final Set<String> names = new java.util.HashSet();
+
+  protected final Project project;
+
+  protected final Configuration classpath;
+
+  @Inject
+  public DefaultWolfExtension(Project project, Configuration classpath) {
+    this.project = project;
+    this.classpath = classpath;
+  }
+
+  public void awooo(String name, Action<? super WolfSpec> action) {
+    if (!this.names.add(name)) {
+      throw new org.gradle.api.GradleException(String.format("An awooo definition with name '%s' was already created", name));
+    }
+    WolfSpec spec = this.project.getObjects().newInstance(WolfSpec.class);
+    this.configureSpec(spec);
+    action.execute(spec);
+    TaskProvider<? extends WolfTask> task = this.createWolfTask(name, new WolfTaskConfigurator(spec, this.classpath));
+  }
+
+  TaskProvider<? extends WolfTask> createWolfTask(String name, Action<WolfTask> configurator) {
+    return this.project.getTasks().register(name, WolfTask.class, configurator);
+  }
+
+  protected void configureSpec(WolfSpec spec) {
+    spec.getAge().convention(1);
+  }
+
+  protected static class WolfTaskConfigurator implements Action<WolfTask> {
+    WolfSpec spec;
+
+    Configuration classpath;
+
+    WolfTaskConfigurator(WolfSpec spec, Configuration classpath) {
+      this.spec = spec;
+      this.classpath = classpath;
+    }
+
+    public void execute(WolfTask arg1) {
+      arg1.getClasspath().from(this.classpath);
+      arg1.setDescription("Configure the awooo");
+      arg1.getSlogan().convention(this.spec.getSlogan());
```

Comment:
The defaults are set in extension, too. This configurator sets all the conventions.